app/dashboard/src/modals/SetupOrganizationAfterSubscribe.tsx

```diff
@@ -0,0 +1,229 @@
+import type { GetText } from '#/providers/TextProvider'
+import * as textProvider from '#/providers/TextProvider'
+
+import * as ariaComponents from '#/components/AriaComponents'
+
+import { Button } from '#/components/AriaComponents'
+import { Result } from '#/components/Result'
+import { Stepper } from '#/components/Stepper'
+import * as backendModule from '#/services/Backend'
+
+// =================
+// === Constants ===
+// =================
+
+const PLANS_TO_SPECIFY_ORG_NAME = [backendModule.Plan.team, backendModule.Plan.enterprise]
+
+// ================================
+// === SetOrganizationNameModal ===
+// ================================
+
+/** Modal for setting the organization name.
+ * Shows up when the user is on the team plan and the organization name is the default. */
+export function SetupOrganizationAfterSubscribe() {
+  const { getText } = textProvider.useText()
+
+  const backend = backendProvider.useRemoteBackendStrict()
+  const { session } = authProvider.useAuth()
+
+  const user = session != null && 'user' in session ? session.user : null
+  const userIsAdmin = user?.isOrganizationAdmin ?? false
+  const userId = user?.userId ?? null
+  const userPlan = user?.plan ?? backendModule.Plan.free
+
+  const { data: organizationName } = useSuspenseQuery({
+    queryKey: ['organization', userId],
+    queryFn: () => backend.getOrganization().catch(() => null),
+    staleTime: Infinity,
+    select: (data) => data?.name ?? '',
+  })
+
+  const { data: hasUserGroups } = useSuspenseQuery({
+    queryKey: ['userGroups', userId],
+    queryFn: () => backend.listUserGroups().catch(() => null),
+    staleTime: Infinity,
+    select: (data) => data?.length ?? 0,
+  })
+
+  const updateOrganization = useMutation(
+    backendMutationOptions(backend, 'updateOrganization', {
+      meta: { invalidates: [['organization', userId]] },
+    }),
+  )
+  const createDefaultUserGroup = useMutation(
+    backendMutationOptions(backend, 'createUserGroup', {
+      meta: { invalidates: [['userGroups', userId]] },
+    }),
+  )
+
+  const shouldSetOrgName = PLANS_TO_SPECIFY_ORG_NAME.includes(userPlan) && organizationName === ''
+  const shouldSetDefaultUserGroup =
+    PLANS_TO_SPECIFY_ORG_NAME.includes(userPlan) && hasUserGroups === 0
+  const shouldShowModal = userIsAdmin && (shouldSetOrgName || shouldSetDefaultUserGroup)
+
+  const { stepperState } = Stepper.useStepperState({
+    steps: 3,
+    defaultStep: 0,
+  })
+
+  return (
+    <>
+      <ariaComponents.Dialog
+        title={getText('setupOrganization')}
+        isDismissable={false}
+        isKeyboardDismissDisabled
+        hideCloseButton
+        size="xxxlarge"
+        padding="xlarge"
+        modalProps={{ isOpen: shouldShowModal }}
+      >
+        <Stepper
+          state={stepperState}
+          renderStep={(props) => (
+            <>
+              {props.index === 0 && <Stepper.Step {...props} title={getText('intro')} />}
```

Comment:
i really don't know about conditional rendering based on `props.index` and `currentStep` especially if there is no shared rendering at all. i'd prefer something more like RAC perhaps
```tsx
<Stepper>
  <Step index={0 /* or even better (?) injected based on child index in the array */}>
    <StepHeader title={getText('a')} />
    <StepBody>
      <Result {...whatever} />
    </StepBody>
  </Step>
</Stepper>

Reply:
Well, yeah, I agree we need more declarative API for that, Not sure the API you described is the best though. Unlike the RAC, we're building a design system that matches our needs, so we don't need to invest more than needed atm into the customization. Even though the proposed solution is flexible, it lacks some advantages we now have:
1. It's not type-safe, we can't tell the user what we expect.
2. it's verbose.
3. It's harder to optimize - control on what to render is yielded to the parent component (we can't skip rendering for offscreen steps)

I'm seeing it a bit differently though and I made a step towards it in the MFA pr:

```tsx
const {state} = Stepper.useStepperState({
  items: Array<T>,
  current: number,
})
 
 // Lazy render
<Stepper state={state} renderStep={({item, ...stepProps}) => (<Stepper.Step {...stepProps}>{item}</Stepper.Step>)}>
  {({item, ...stepProps}) => <Stepper.StepContent {...stepProps}>{item}</Stepper.StepContent>}
</Stepper>
```

or

```tsx
const {state} = Stepper.useStepperState({
  items: Array<T>,
  current: number,
})

// Simple render, all items are rendered
<Stepper state={state} renderStep={({item, ...stepProps}) => (<Stepper.Step {...stepProps}>{item}</Stepper.Step>)}>
  {state.items.map((item, index) => (
    <Stepper.StepContent key={index} index={index} title={item.title}>
      {item.content}
    </Stepper.StepContent>
  ))}
</Stepper>
```

WDYT?

Reply:
i dunno... honestly my main issue is that the header is separate from the content
i guess using JSX for everything *is* going too far, but maybe we could have an array of steps or something? (`{ header: JSX.Element, content: (props) => JSX.Element }`)